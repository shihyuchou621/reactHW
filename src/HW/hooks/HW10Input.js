import React from 'react';

export function HW10Input(props) {
  const {
    name,
    rate,
    money,
    dollar,
    onChange,
  } = props;

  const value = dollar === name ? money: money * rate;

  return (
    <div>
      {name}
      <input
        data-testid="HW10Input"
        name={name}
        value={value}
        onChange={onChange}
      />
    </div>
  );
}
