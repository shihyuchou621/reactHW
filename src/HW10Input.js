import React, { Component } from 'react'

/** props.name, rate, onChange */
export class HW10Input extends Component {
  render() {

    const {
      name,
      rate,
      money,
      dollar,
      onChange,
    } = this.props;

    const value = dollar === name ? money: money * rate;

    return (
      <div>
        <input
          data-testid="HW10Input"
          name={name}
          value={value}
          onChange={onChange}
        />
      </div>
    )
  }
}
