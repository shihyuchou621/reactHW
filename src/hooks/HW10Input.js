import React from 'react';

export function HW10Input(props) {
  const {
    name,
    rate,
    money,
    dollar,
    onChange,
  } = props;

  const value = dollar === name ? money: money * rate;

  return (
    <div>
      <input
        data-testid="HW10Input"
        name={name}
        value={value}
        onChange={onChange}
      />
    </div>
  );
}
