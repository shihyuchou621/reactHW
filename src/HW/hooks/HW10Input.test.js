import React from 'react';
import {
  render,
  cleanup,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW10Input from './HW10Input';

afterEach(cleanup);

test('onChange should work', () => {
  const onChange = jest.fn(e => e.target.value);
  const { container, getByTestId } = render(
    <HW10Input
      name="usd"
      rate={30}
      money={1}
      dollar="usd"
      onChange={onChange}
    />
  );
  userEvent.type(getByTestId('HW10Input'), '10');

  // !!!!↓找不到value，result是空陣列，其他屬性也沒找到
  console.log(onChange.mock.results[0]);
  // expect(onChange.mock.results[0].value).toBe("10");

  expect(getByTestId('HW10Input').value).toBe("1");
  expect(container).toMatchSnapshot();
});

test('should have correct value', () => {
  const onChange = jest.fn();
  const { container, getByTestId } = render(
    <HW10Input
      name="usd"
      rate={1/30}
      money={90}
      value={0}
      dollar="ntd"
      onChange={onChange}
    />
  );
  expect(getByTestId('HW10Input').value).toBe("3");
  expect(container).toMatchSnapshot();
});
