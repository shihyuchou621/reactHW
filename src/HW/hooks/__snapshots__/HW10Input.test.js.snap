// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`onChange should work 1`] = `
<div>
  <div>
    usd
    <input
      data-testid="HW10Input"
      name="usd"
      value="1"
    />
  </div>
</div>
`;

exports[`should have correct value 1`] = `
<div>
  <div>
    usd
    <input
      data-testid="HW10Input"
      name="usd"
      value="3"
    />
  </div>
</div>
`;
